src/lib/discord/commands/admin.ts

```diff
@@ -0,0 +1,215 @@
+import { SlashCommandBuilder } from '@discordjs/builders'
+import { faker } from '@faker-js/faker'
+import { EmbedBuilder } from 'discord.js'
+import { getUserAccess } from '$discord/get-user-access'
+import { prisma } from '$lib/db'
+import {
+  postDiscussion,
+  postAnswer,
+  markAnswered,
+  lockDiscussion,
+} from '$lib/github/queries'
+import { repositoriesWithDiscussions as repositories } from './_repositories'
+import { isThreadWithinHelpChannel } from '../support'
+import type {
+  GuildMember,
+  Message,
+  ChatInputCommandInteraction,
+  InteractionReplyOptions,
+  ThreadChannel,
+} from 'discord.js'
+
+const userIdToUsername = new Map<string, user>()
+
+// const iconMap = new Map<string, string>([
```

Comment:
I want to display a small circle next to the user's role that matches the Discord default role color but can't for the life of me get it to work

Reply:
```
<img width="20px" height="20px" src="https://raw.githubusercontent.com/esauerbo1/Images/main/Bot.svg" />
```

<img width="20px" height="20px" src="https://raw.githubusercontent.com/esauerbo1/Images/main/Bot.svg" />

when the images are already created we can use an `img` tag, BUT it would be neat to dynamically generate this file from an endpoint

Reply:
agreed... but I couldn't find anything on the api for it and it's also not part of the role object. totally not something we have to include I just thought it could be cool

Reply:
I think this should work!

```ts
// src/routes/api/p/color/[code].ts
import type { RequestHandler } from '@sveltejs/kit'

const svg = (color: string) => `
  <svg 
    fill="${color}"
    xmlns="http://www.w3.org/2000/svg"
    xmlns:xlink= "http://www.w3.org/1999/xlink"
  >
    <circle cx="50" cy="50" r="40" />
  </svg>
`

export const GET: RequestHandler = ({ params }) => {
  const { code } = params
  const color = `#${code.replace(/\.svg$/, '')}`
  return {
    headers: {
      // store for 365 days
      'cache-control': import.meta.env.PROD
        ? 'public, max-age=31536000, immutable'
        : 'no-cache',
      // store for 30 days
      // 'cache-control': 'public, max-age=2592000, immutable',
      'content-type': 'image/svg+xml',
    },
    body: svg(color),
  }
}
```

<img width="209" alt="image" src="https://user-images.githubusercontent.com/5033303/183171249-cdd1de09-97b4-4c70-8b6d-537cd5b950eb.png">

Reply:
amazing!!